app/uk/gov/hmrc/exports/scheduler/jobs/emails/EmailCancellationValidator.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.scheduler.jobs.emails
+
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.exports.models.declaration.submissions.SubmissionStatus.{CANCELLED, CLEARED, REJECTED}
+import uk.gov.hmrc.exports.models.emails.SendEmailDetails
+import uk.gov.hmrc.exports.repositories.NotificationRepository
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+private[emails] class EmailCancellationValidator @Inject()(notificationRepository: NotificationRepository) {
+
+  private val statusesCancellingEmailSending = Set(REJECTED, CLEARED, CANCELLED)
+
+  def isEmailSendingCancelled(sendEmailDetails: SendEmailDetails)(implicit ec: ExecutionContext): Future[Boolean] =
+    notificationRepository.findNotificationsByMrn(sendEmailDetails.mrn).map { notifications =>
+      val currentDmsDocNotification = notifications
+        .find(_.id == sendEmailDetails.notificationId)
```

Comment:
This will find the first DMSDOC notification, but what about subsequent DMSDOC notifications (that might have been sent later)? Should you not get the last DMSDOC notification for this MRN?

Reply:
The other DMSDOC would have it's own WorkItem created. This process relates to specific DMSDOC, not to all DMSDOCs for given MRN. So if one DMSDOC has been cancelled, there might still be another DMSDOC received for this MRN and an email should be sent. And the other DMSDOC will be checked separately.